drivers/modem/gsm_ppp.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2021 G-Technologies Sdn. Bhd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef GSM_PPP_H
+#define GSM_PPP_H
+
+#include <device.h>
+#include "modem_context.h"
+
+#define HAS_PWR_SRC DT_INST_NODE_HAS_PROP(0, power_src_gpios)
+#define HAS_PWR_KEY DT_INST_NODE_HAS_PROP(0, power_key_gpios)
+
+#if HAS_PWR_SRC || HAS_PWR_KEY
+/* pin settings */
+enum mdm_control_pins {
+#if HAS_PWR_SRC
+	MDM_PWR_SRC,
+#endif
+#if HAS_PWR_KEY
+	MDM_PWR_KEY,
+#endif
+};
+
+/* Modem pins - Power, Reset & others. */
+static struct modem_pin modem_pins[] = {
+#if HAS_PWR_SRC
+	/* MDM_POWER SUPPLY */
+	MODEM_PIN(DT_INST_GPIO_LABEL(0, power_src_gpios), DT_INST_GPIO_PIN(0, power_src_gpios),
+		  DT_INST_GPIO_FLAGS(0, power_src_gpios) | GPIO_OUTPUT_LOW),
+#endif
+
+#if HAS_PWR_KEY
+	/* MDM_POWER_KEY */
+	MODEM_PIN(DT_INST_GPIO_LABEL(0, power_key_gpios), DT_INST_GPIO_PIN(0, power_key_gpios),
+		  DT_INST_GPIO_FLAGS(0, power_key_gpios) | GPIO_OUTPUT_LOW),
+#endif
+};
+#endif
+
+/**
+ * @brief  Disable power source of the modem.
+ *
+ * @param  ctx: modem_context struct
+ *
+ * @retval None.
+ */
+void disable_power_source(struct modem_context *ctx)
+{
+#if HAS_PWR_SRC
+	modem_pin_write(ctx, MDM_PWR_SRC, 0);
+#endif
+}
+
+/**
+ * @brief  Enable power source of the modem.
+ *
+ * @param  ctx: modem_context struct
+ *
+ * @retval None.
+ */
+void enable_power_source(struct modem_context *ctx)
```

Comment:
first of all these aren't namespaced appropriately.

Reply:
also they lack the inline qualifier?

Reply:
Not very familiar with "inline", let me have a look online.